src/odemis/dataio/hdf5.py

```diff
@@ -1200,132 +1030,72 @@ def _add_image_metadata(group, image, mds):
+    set_status(gp, "LinearPolarizer", st_poslinpol, poslinpol)
+    # streak camera
+    set_status(gp, "TimeRange", st_timeRange, timeRange)
+    set_status(gp, "MCPGain", st_MCPGain, MCPGain)
+    set_status(gp, "StreakMode", st_streakMode, streakMode)
+    set_status(gp, "TriggerDelay", st_triggerDelay, triggerDelay)
+    set_status(gp, "TriggerRate", st_triggerRate, triggerRate)
+
+
+def get_status(status_list, value_list, md, MD_key, default=True):
+    """
+
+    """
+    if MD_key in md:
+        value_list.append(md[MD_key])
+        status_list.append(ST_REPORTED)
+    else:
+        if default:
+            value_list.append("")
         else:
-            triggerRate.append("")
-            st_triggerRate.append(ST_INVALID)
-
-    if not all(st == ST_INVALID for st in st_pp):
-        gp["PolePosition"] = pp
-        _h5svi_set_state(gp["PolePosition"], st_pp)
-    if not all(st == ST_INVALID for st in st_xm):
-        gp["XMax"] = xm
-        _h5svi_set_state(gp["XMax"], st_xm)
-    if not all(st == ST_INVALID for st in st_hd):
-        gp["HoleDiameter"] = hd
-        _h5svi_set_state(gp["HoleDiameter"], st_hd)
-    if not all(st == ST_INVALID for st in st_fd):
-        gp["FocusDistance"] = fd
-        _h5svi_set_state(gp["FocusDistance"], st_fd)
-    if not all(st == ST_INVALID for st in st_pf):
-        gp["ParabolaF"] = pf
-        _h5svi_set_state(gp["ParabolaF"], st_pf)
+            if MD_key is model.MD_AR_POLE:
+                value_list.append((0, 0))
+            else:
+                value_list.append(0)
+        status_list.append(ST_INVALID)
 
-    # polarization analyzer
-    if not all(st == ST_INVALID for st in st_pol):
-        gp["Polarization"] = pol
-        _h5svi_set_state(gp["Polarization"], st_pol)
-    if not all(st == ST_INVALID for st in st_posqwp):
-        gp["QuarterWavePlate"] = posqwp
-        _h5svi_set_state(gp["QuarterWavePlate"], st_posqwp)
-    if not all(st == ST_INVALID for st in st_poslinpol):
-        gp["LinearPolarizer"] = poslinpol
-        _h5svi_set_state(gp["LinearPolarizer"], st_poslinpol)
 
-    # streak camera
-    if not all(st == ST_INVALID for st in st_timeRange):
-        gp["TimeRange"] = timeRange
-        _h5svi_set_state(gp["TimeRange"], st_timeRange)
-    if not all(st == ST_INVALID for st in st_MCPGain):
-        gp["MCPGain"] = MCPGain
-        _h5svi_set_state(gp["MCPGain"], st_MCPGain)
-    if not all(st == ST_INVALID for st in st_streakMode):
-        gp["StreakMode"] = streakMode
-        _h5svi_set_state(gp["StreakMode"], st_streakMode)
-    if not all(st == ST_INVALID for st in st_triggerDelay):
-        gp["TriggerDelay"] = triggerDelay
-        _h5svi_set_state(gp["TriggerDelay"], st_triggerDelay)
-    if not all(st == ST_INVALID for st in st_triggerRate):
-        gp["TriggerRate"] = triggerRate
-        _h5svi_set_state(gp["TriggerRate"], st_triggerRate)
+def set_status(gp, name, status_list, value_list):
```

Comment:
Let's call this `set_metadata` or `write_metadata`? (where "metadata" is whichever word you've picked for the previous function).

Please, also fill the docstring too here :-D